srcts/src/components/tooltip.ts

```diff
@@ -0,0 +1,181 @@
+import { nothing } from "lit";
+import { property } from "lit/decorators.js";
+import { LightElement } from "./webcomponents/_lightElement";
+import type { HtmlDep } from "./_utils";
+import type { Tooltip as TooltipType } from "bootstrap";
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+const Tooltip = (
+  window.bootstrap ? window.bootstrap.Tooltip : class {}
+) as typeof TooltipType;
+
+type TooltipOptions = {
+  title: TooltipType.Options["title"];
+  placement: TooltipType.Options["placement"];
+  html: boolean;
+  sanitize: boolean;
+};
+
+type ToggleMessage = {
+  method: "toggle";
+  value: "hide" | "show" | "toggle";
+};
+
+type UpdateMessage = {
+  method: "update";
+  title?: { html: string; deps: HtmlDep[] };
+};
+
+type MessageData = ToggleMessage | UpdateMessage;
+
+type TooltipTypeWithTip = TooltipType & {
+  tip: HTMLElement | null;
+};
+
+export class BslibTooltip extends LightElement {
+  static tagName = "bslib-tooltip";
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  _tooltip!: TooltipType;
+
+  @property({ type: String }) placement: TooltipOptions["placement"] = "auto";
+  @property({ type: String }) options = "{}";
+
+  get allOptions(): TooltipOptions {
+    const opts = JSON.parse(this.options);
+    return {
+      title: this.title,
+      placement: this.placement,
+      // Bootstrap defaults to false, but we have our own HTML escaping
+      html: true,
+      sanitize: true,
+      ...opts,
+    };
+  }
+
+  get title(): string {
+    return (this.children[0] as Element).innerHTML;
+  }
+
+  constructor() {
+    super();
+    this._onShown = this._onShown.bind(this);
+    this._onHidden = this._onHidden.bind(this);
+    this.style.display = "contents";
+  }
+
+  connectedCallback(): void {
+    super.connectedCallback();
+    this.reference.setAttribute("data-bs-toggle", "tooltip");
+    this._tooltip = new Tooltip(this.reference, this.allOptions);
+
+    this.reference.addEventListener("shown.bs.tooltip", this._onShown);
+    this.reference.addEventListener("hidden.bs.tooltip", this._onHidden);
+  }
+
+  disconnectedCallback(): void {
+    this.reference.removeEventListener("shown.bs.tooltip", this._onShown);
+    this.reference.removeEventListener("hidden.bs.tooltip", this._onHidden);
+
+    super.disconnectedCallback();
+  }
+
+  render(): typeof nothing {
+    return nothing;
+  }
+
+  // Find an Element to use as the reference for the tooltip
+  //
+  // TODO: In the future, it'd be nice if the reference was a virtual element (defining)
+  // a rectangle around `this.childNodes` instead of just the last HTMLElement.
+  // As of today, bootstrap.Tooltip doesn't seem to support floating-ui's virtual elements,
+  // (but that should change in Bootstrap v6 https://github.com/twbs/bootstrap/pull/36683)
+  get reference(): Element {
```

Comment:
Nit: I think we can find a better name than `reference`, since it's hard to tell if you mean the trigger element or a "reference" for the tooltip content. Maybe `triggerElement`?